### ISSUE TITLE
Square brackets break Emmet in TypeScript files

### ISSUE THREAD

**Issue by marcushellberg** [microsoft/vscode#161431]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.71.2
- OS Version: MacOS 12.5.1

Steps to Reproduce:

1. Enable Emmet for TypeScript files, using the `html` language 
2. Type a snippet with an attribute, like `a[href]`. Emmet does not expand the snippet. 

Emmet behaves as expected in HTML files. E used to work prior to the VS Code update that came out in mid-May 2022. 

Video showing unexpected behavior in a TypeScript file: 
https://user-images.githubusercontent.com/883698/191588848-4f666b06-0d4e-49a1-8ac5-1fc59f34df36.mp4

Video showing expected behavior in an HTML file. 
https://user-images.githubusercontent.com/883698/191589152-29a0712a-b7c7-49df-9002-0c573fc07680.mp4

**Comment by rzhao271** [comment]
This is the reverse of https://github.com/microsoft/vscode/issues/138461.
Does enabling `emmet.triggerExpandOnTab` and pressing tab work?

**Comment by marcushellberg** [comment]
Yes, the expansion works. It doesn't show the autocomplete dropdown, but the actual expansion does work.
